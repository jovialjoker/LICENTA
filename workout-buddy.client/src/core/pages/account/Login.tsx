import {
  Flex,
  Box,
  FormControl,
  FormLabel,
  Input,
  Checkbox,
  Stack,
  Link,
  Button,
  Text,
  Heading,
  useColorModeValue,
} from "@chakra-ui/react";
import { useState } from "react";
import { accountActions } from "../../../store/reducers/account";
import axios from "axios";
import { useDispatch } from "react-redux";

const loginModelInitialState = {
  email: "",
  password: "",
  areCredentialsInvalid: false,
  isDisabled: false,
};

export default function Login() {
  const dispatcher = useDispatch();
  const [authError, setAuthError] = useState<string | null>(null);
  const [loginModel, setLoginModel] = useState(loginModelInitialState);

  const submitHandler = async (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    e.preventDefault();
    try {
      const res = await axios({
        method: "post",
        url: "https://localhost:7132/UserAccount/login",
        data: loginModel,
      });
      dispatcher(accountActions.login(res.data));

      window.location.href = "/";
    } catch (e: any) {
      setAuthError("Your email/password combination is not correct!");
    }
  };

  return (
    <Flex
      align={"center"}
      justify={"center"}
      bg={useColorModeValue("white", "gray.800")}
    >
      <Stack spacing={8} mx={"auto"} maxW={"lg"} py={12} px={6}>
        <Stack align={"center"}>
          <Heading fontSize={"4xl"}>Sign in to your account</Heading>
        </Stack>
        <Box
          rounded={"lg"}
          bg={useColorModeValue("gray.50", "gray.700")}
          boxShadow={"lg"}
          p={8}
          borderColor="lightPallette.primary.main"
          borderWidth="2px"
        >
          <Stack spacing={4}>
            {authError && <Text color="red.700">{authError}</Text>}
            <FormControl id="email">
              <FormLabel>Email address</FormLabel>
              <Input
                data-testid="email-login-input"
                type="email"
                value={loginModel.email}
                onChange={(e) => {
                  setAuthError(null);
                  setLoginModel({ ...loginModel, email: e.target.value });
                }}
              />
            </FormControl>
            <FormControl id="password">
              <FormLabel>Password</FormLabel>
              <Input
                data-testid="password-login-input"
                type="password"
                value={loginModel.password}
                onChange={(e) => {
                  setAuthError(null);
                  setLoginModel({ ...loginModel, password: e.target.value });
                }}
              />
            </FormControl>
            <Stack spacing={10}>
              <Stack
                direction={{ base: "column", sm: "row" }}
                align={"start"}
                justify={"space-between"}
              >
                <Checkbox checked={false} data-testid="remember-login-checkbox">
                  Remember me
                </Checkbox>
                <Link
                  color={useColorModeValue(
                    "lightPallette.accent.main",
                    "darkPallette.accent.main"
                  )}
                >
                  Forgot password?
                </Link>
              </Stack>
              <Button
                colorScheme={useColorModeValue(
                  "lightPallette.primary",
                  "darkPallette.primary"
                )}
                onClick={submitHandler}
              >
                Sign in
              </Button>
            </Stack>
          </Stack>
        </Box>
      </Stack>
    </Flex>
  );
}
