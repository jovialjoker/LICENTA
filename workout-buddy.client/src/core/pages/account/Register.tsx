import {
  Flex,
  Box,
  FormControl,
  FormLabel,
  Input,
  InputGroup,
  HStack,
  InputRightElement,
  Stack,
  Button,
  Heading,
  Text,
  useColorModeValue,
  Link,
  FormErrorMessage,
} from "@chakra-ui/react";
import { useState } from "react";
import { ViewIcon, ViewOffIcon } from "@chakra-ui/icons";
import { useDispatch } from "react-redux";
import { accountActions } from "../../../store/reducers/account";
import axios from "axios";

const registerModelInitialState = {
  name: "",
  username: "",
  email: "",
  passwordString: "",
  birthDay: "",
  weight: 7,
};

export default function Register() {
  const dispatcher = useDispatch();
  const [registerModel, setRegisterModel] = useState(registerModelInitialState);
  const [showPassword, setShowPassword] = useState(false);
  const [authError, setAuthError] = useState<
    { propertyName: string; errorMessage: string }[]
  >([]);

  const submitHandler = async (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    e.preventDefault();
    try {
      const res = await axios({
        method: "post",
        url: "http://localhost:8082/UserAccount/register",
        data: registerModel,
      });

      dispatcher(accountActions.register(res.data));

      window.location.href = "/";
    } catch (e: any) {
      const { data } = e.response;
      console.log(data);
      setAuthError(data);
    }
  };

  return (
    <Flex
      align={"center"}
      justify={"center"}
      bg={useColorModeValue("white", "gray.800")}
    >
      <Stack spacing={8} mx={"auto"} maxW={"lg"} py={12} px={6}>
        <Stack align={"center"}>
          <Heading fontSize={"4xl"} textAlign={"center"}>
            Sign up
          </Heading>
        </Stack>
        <Box
          rounded={"lg"}
          bg={useColorModeValue("white", "gray.700")}
          boxShadow={"lg"}
          p={8}
          borderColor="lightPallette.primary.main"
          borderWidth="2px"
        >
          <Stack spacing={4}>
            <HStack>
              <Box>
                <FormControl id="firstName" isRequired>
                  <FormLabel>Name</FormLabel>
                  <Input
                    data-testid="register-name"
                    type="text"
                    value={registerModel.name}
                    onChange={(e) =>
                      setRegisterModel({
                        ...registerModel,
                        name: e.target.value,
                      })
                    }
                  />
                  <Text color="red.500" fontSize="13px">
                    {
                      authError.find((err) => err.propertyName === "Name")
                        ?.errorMessage
                    }
                  </Text>
                </FormControl>
              </Box>
              <Box>
                <FormControl id="lastName" isRequired>
                  <FormLabel>Username</FormLabel>
                  <Input
                    data-testid="register-username"
                    type="text"
                    value={registerModel.username}
                    onChange={(e) =>
                      setRegisterModel({
                        ...registerModel,
                        username: e.target.value,
                      })
                    }
                  />
                  <Text color="red.500" fontSize="13px">
                    {
                      authError.find((err) => err.propertyName === "Username")
                        ?.errorMessage
                    }
                  </Text>
                </FormControl>
              </Box>
            </HStack>
            <FormControl id="email" isRequired>
              <FormLabel>Email address</FormLabel>
              <Input
                data-testid="register-email"
                type="email"
                value={registerModel.email}
                onChange={(e) =>
                  setRegisterModel({ ...registerModel, email: e.target.value })
                }
              />
              <Text color="red.500" fontSize="13px">
                {
                  authError.find((err) => err.propertyName === "Email")
                    ?.errorMessage
                }
              </Text>
            </FormControl>
            <FormControl id="password" isRequired>
              <FormLabel>Password</FormLabel>
              <InputGroup>
                <Input
                  data-testid="register-password"
                  type={showPassword ? "text" : "password"}
                  value={registerModel.passwordString}
                  onChange={(e) =>
                    setRegisterModel({
                      ...registerModel,
                      passwordString: e.target.value,
                    })
                  }
                />
                <InputRightElement h={"full"}>
                  <Button
                    variant={"ghost"}
                    onClick={() =>
                      setShowPassword((showPassword) => !showPassword)
                    }
                  >
                    {showPassword ? <ViewIcon /> : <ViewOffIcon />}
                  </Button>
                </InputRightElement>
              </InputGroup>
              <Text color="red.500" fontSize="13px">
                {
                  authError.find((err) => err.propertyName === "PasswordString")
                    ?.errorMessage
                }
              </Text>
            </FormControl>
            <FormControl id="birthdate" isRequired>
              <FormLabel>Birth Date</FormLabel>
              <Input
                data-testid="register-birthdate"
                type="datetime-local"
                value={registerModel.birthDay}
                onChange={(e) =>
                  setRegisterModel({
                    ...registerModel,
                    birthDay: e.target.value,
                  })
                }
              ></Input>
              <Text color="red.500" fontSize="13px">
                {
                  authError.find((err) => err.propertyName === "BirthDay")
                    ?.errorMessage
                }
              </Text>
            </FormControl>
            <FormControl id="weight" isRequired>
              <FormLabel>Weight</FormLabel>
              <Input
                data-testid="register-weight"
                type="number"
                value={registerModel.weight}
                onChange={(e) =>
                  setRegisterModel({
                    ...registerModel,
                    weight: +e.target.value,
                  })
                }
              ></Input>
              <Text color="red.500" fontSize="13px">
                {
                  authError.find((err) => err.propertyName === "Weight")
                    ?.errorMessage
                }
              </Text>
            </FormControl>
            <Stack spacing={10} pt={2}>
              <Button
                loadingText="Submitting"
                size="lg"
                colorScheme={useColorModeValue(
                  "lightPallette.primary",
                  "darkPallette.primary"
                )}
                onClick={submitHandler}
              >
                Sign up
              </Button>
            </Stack>
            <Stack pt={6}>
              <Text align={"center"}>
                Already a user?{" "}
                <Link color={"lightPallette.primary.500"} href="/login">
                  Login
                </Link>
              </Text>
            </Stack>
          </Stack>
        </Box>
      </Stack>
    </Flex>
  );
}
