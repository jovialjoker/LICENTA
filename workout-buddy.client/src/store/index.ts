import { configureStore } from "@reduxjs/toolkit";
import { accountReducer } from "./reducers/account";
import { useDispatch } from "react-redux";

const store = configureStore({
  reducer: {
    account: accountReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;

export default store;
