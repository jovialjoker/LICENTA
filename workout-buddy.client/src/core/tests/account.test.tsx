import { render } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import Login from "../pages/account/Login";
import Register from "../pages/account/Register";
import { Provider } from "react-redux";
import store from "../../store";

test("renders login with initial state", () => {
  render(
    <Provider store={store}>
      <Login />
    </Provider>
  );
  const emailInput = screen.getByTestId("email-login-input");
  const passwordInput = screen.getByTestId("password-login-input");
  const rememberCheckbox = screen.getByTestId(
    "remember-login-checkbox"
  ) as HTMLInputElement;
  expect(emailInput.textContent).toBe("");
  expect(passwordInput.textContent).toBe("");
  expect(rememberCheckbox.checked).toBe(false);
});

test("renders register with initial state", () => {
  render(
    <Provider store={store}>
      <Register />
    </Provider>
  );
  const name = screen.getByTestId("register-name");
  const username = screen.getByTestId("register-username");
  const email = screen.getByTestId("register-email");
  const password = screen.getByTestId("register-password");
  const birthdate = screen.getByTestId("register-birthdate");
  const weight = screen.getByTestId("register-weight") as HTMLInputElement;
  expect(name.textContent).toBe("");
  expect(username.textContent).toBe("");
  expect(email.textContent).toBe("");
  expect(password.textContent).toBe("");
  expect(birthdate.textContent).toBe("");
  expect(weight.value).toBe("7");
});

test("check login snapshot", () => {
  const component = render(
    <Provider store={store}>
      <Login />
    </Provider>
  );

  expect(component).toMatchSnapshot();
});

test("check register snapshot", () => {
  const component = render(
    <Provider store={store}>
      <Register />
    </Provider>
  );

  expect(component).toMatchSnapshot();
});
