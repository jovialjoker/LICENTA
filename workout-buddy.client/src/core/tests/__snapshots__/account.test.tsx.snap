// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check login snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-18308nr"
      >
        <div
          class="chakra-stack css-100xlw7"
        >
          <div
            class="chakra-stack css-tl3ftk"
          >
            <h2
              class="chakra-heading css-1gjuw9l"
            >
              Sign in to your account
            </h2>
          </div>
          <div
            class="css-lfqhk9"
          >
            <div
              class="chakra-stack css-1kkt86i"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="email"
                  id="email-label"
                >
                  Email address
                </label>
                <input
                  class="chakra-input css-0"
                  data-testid="email-login-input"
                  id="email"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="password"
                  id="password-label"
                >
                  Password
                </label>
                <input
                  class="chakra-input css-0"
                  data-testid="password-login-input"
                  id="password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="chakra-stack css-bco1gb"
              >
                <div
                  class="chakra-stack css-1ujp649"
                >
                  <label
                    class="chakra-checkbox css-1elxtiq"
                    data-testid="remember-login-checkbox"
                  >
                    <input
                      class="chakra-checkbox__input"
                      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                      type="checkbox"
                      value=""
                    />
                    <span
                      aria-hidden="true"
                      class="chakra-checkbox__control css-dnty2r"
                    />
                    <span
                      class="chakra-checkbox__label css-1oeb2oe"
                    >
                      Remember me
                    </span>
                  </label>
                  <a
                    class="chakra-link css-1lhnvgm"
                  >
                    Forgot password?
                  </a>
                </div>
                <button
                  class="chakra-button css-4xx2wk"
                  type="button"
                >
                  Sign in
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-18308nr"
    >
      <div
        class="chakra-stack css-100xlw7"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <h2
            class="chakra-heading css-1gjuw9l"
          >
            Sign in to your account
          </h2>
        </div>
        <div
          class="css-lfqhk9"
        >
          <div
            class="chakra-stack css-1kkt86i"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="email"
                id="email-label"
              >
                Email address
              </label>
              <input
                class="chakra-input css-0"
                data-testid="email-login-input"
                id="email"
                type="email"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="password"
                id="password-label"
              >
                Password
              </label>
              <input
                class="chakra-input css-0"
                data-testid="password-login-input"
                id="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="chakra-stack css-bco1gb"
            >
              <div
                class="chakra-stack css-1ujp649"
              >
                <label
                  class="chakra-checkbox css-1elxtiq"
                  data-testid="remember-login-checkbox"
                >
                  <input
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-dnty2r"
                  />
                  <span
                    class="chakra-checkbox__label css-1oeb2oe"
                  >
                    Remember me
                  </span>
                </label>
                <a
                  class="chakra-link css-1lhnvgm"
                >
                  Forgot password?
                </a>
              </div>
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                Sign in
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`check register snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-18308nr"
      >
        <div
          class="chakra-stack css-100xlw7"
        >
          <div
            class="chakra-stack css-tl3ftk"
          >
            <h2
              class="chakra-heading css-5efzzv"
            >
              Sign up
            </h2>
          </div>
          <div
            class="css-16n56wr"
          >
            <div
              class="chakra-stack css-1kkt86i"
            >
              <div
                class="chakra-stack css-1igwmid"
              >
                <div
                  class="css-0"
                >
                  <div
                    class="chakra-form-control css-0"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-uvho5g"
                      for="firstName"
                      id="firstName-label"
                    >
                      Name
                      <span
                        aria-hidden="true"
                        class="chakra-form__required-indicator css-0"
                        role="presentation"
                      >
                        *
                      </span>
                    </label>
                    <input
                      aria-required="true"
                      class="chakra-input css-0"
                      data-testid="register-name"
                      id="firstName"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <div
                    class="chakra-form-control css-0"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-uvho5g"
                      for="lastName"
                      id="lastName-label"
                    >
                      Username
                      <span
                        aria-hidden="true"
                        class="chakra-form__required-indicator css-0"
                        role="presentation"
                      >
                        *
                      </span>
                    </label>
                    <input
                      aria-required="true"
                      class="chakra-input css-0"
                      data-testid="register-username"
                      id="lastName"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="email"
                  id="email-label"
                >
                  Email address
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  class="chakra-input css-0"
                  data-testid="register-email"
                  id="email"
                  required=""
                  type="email"
                  value=""
                />
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="password"
                  id="password-label"
                >
                  Password
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <div
                  class="chakra-input__group css-bx0blc"
                  data-group="true"
                >
                  <input
                    aria-required="true"
                    class="chakra-input css-0"
                    data-testid="register-password"
                    id="password"
                    required=""
                    type="password"
                    value=""
                  />
                  <div
                    class="chakra-input__right-element css-1aezdzf"
                  >
                    <button
                      class="chakra-button css-4xx2wk"
                      type="button"
                    >
                      <svg
                        class="chakra-icon css-onkibi"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <g
                          fill="currentColor"
                        >
                          <path
                            d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                          />
                          <path
                            d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                          />
                        </g>
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="birthdate"
                  id="birthdate-label"
                >
                  Birth Date
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  class="chakra-input css-0"
                  data-testid="register-birthdate"
                  id="birthdate"
                  required=""
                  type="datetime-local"
                  value=""
                />
              </div>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="weight"
                  id="weight-label"
                >
                  Weight
                  <span
                    aria-hidden="true"
                    class="chakra-form__required-indicator css-0"
                    role="presentation"
                  >
                    *
                  </span>
                </label>
                <input
                  aria-required="true"
                  class="chakra-input css-0"
                  data-testid="register-weight"
                  id="weight"
                  required=""
                  type="number"
                  value="7"
                />
              </div>
              <div
                class="chakra-stack css-1fivd04"
              >
                <button
                  class="chakra-button css-4xx2wk"
                  type="button"
                >
                  Sign up
                </button>
              </div>
              <div
                class="chakra-stack css-r6ry2f"
              >
                <p
                  class="chakra-text css-xi606m"
                >
                  Already a user?
                   
                  <a
                    class="chakra-link css-ufrqji"
                    href="/login"
                  >
                    Login
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-18308nr"
    >
      <div
        class="chakra-stack css-100xlw7"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <h2
            class="chakra-heading css-5efzzv"
          >
            Sign up
          </h2>
        </div>
        <div
          class="css-16n56wr"
        >
          <div
            class="chakra-stack css-1kkt86i"
          >
            <div
              class="chakra-stack css-1igwmid"
            >
              <div
                class="css-0"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="firstName"
                    id="firstName-label"
                  >
                    Name
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-0"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <input
                    aria-required="true"
                    class="chakra-input css-0"
                    data-testid="register-name"
                    id="firstName"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="css-0"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-uvho5g"
                    for="lastName"
                    id="lastName-label"
                  >
                    Username
                    <span
                      aria-hidden="true"
                      class="chakra-form__required-indicator css-0"
                      role="presentation"
                    >
                      *
                    </span>
                  </label>
                  <input
                    aria-required="true"
                    class="chakra-input css-0"
                    data-testid="register-username"
                    id="lastName"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="email"
                id="email-label"
              >
                Email address
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-0"
                data-testid="register-email"
                id="email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="password"
                id="password-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <input
                  aria-required="true"
                  class="chakra-input css-0"
                  data-testid="register-password"
                  id="password"
                  required=""
                  type="password"
                  value=""
                />
                <div
                  class="chakra-input__right-element css-1aezdzf"
                >
                  <button
                    class="chakra-button css-4xx2wk"
                    type="button"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <g
                        fill="currentColor"
                      >
                        <path
                          d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                        />
                        <path
                          d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                        />
                      </g>
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="birthdate"
                id="birthdate-label"
              >
                Birth Date
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-0"
                data-testid="register-birthdate"
                id="birthdate"
                required=""
                type="datetime-local"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="weight"
                id="weight-label"
              >
                Weight
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <input
                aria-required="true"
                class="chakra-input css-0"
                data-testid="register-weight"
                id="weight"
                required=""
                type="number"
                value="7"
              />
            </div>
            <div
              class="chakra-stack css-1fivd04"
            >
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                Sign up
              </button>
            </div>
            <div
              class="chakra-stack css-r6ry2f"
            >
              <p
                class="chakra-text css-xi606m"
              >
                Already a user?
                 
                <a
                  class="chakra-link css-ufrqji"
                  href="/login"
                >
                  Login
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
